agent/engine/dependencygraph/graph.go

```diff
@@ -367,6 +379,18 @@ func containerOrderingDependenciesIsResolved(target *apicontainer.Container,
 	}
 }
 
+func hasDependencyTimedOut(dependOnContainer *apicontainer.Container, dependencyCondition string) bool {
+	switch dependencyCondition {
+	case successCondition, completeCondition, healthyCondition:
+		if !dependOnContainer.GetStartedAt().IsZero() && dependOnContainer.GetStartTimeout() > 0 {
```

Comment:
Can you have both of these conditions follow the same idiom? Either IsZero or comparison is fine.

You can reduce indentation by putting this above the switch:

```
if dependOnContainer.GetStartedAt().IsZero() || dependOnContainer.GetStartTimeout().IsZero()
   return false
}

switch dependencyCondition {
```

Reply:
isZero() checks if the time.Time value is the 0th time for golang. and on the other hand, GetStartTimeout () returns a time duration. That is why, isZero() is not valid for GetStartTimeout (as it returns time.Duration) and >0 cannot be used for GetStartedAt (as it returns time.Time).